# Quality of Life Dataset Analysis

### Abstract

### Dataset description

The dataset belongs to the socio-economic domain and contains quality of life indicators for different countries between 2015 and 2024.

It includes information related to purchasing power, safety, healthcare, cost of living, pollution, climate conditions, and overall quality of life.

In [2]:
import pandas as pd

In [3]:
data = pd.read_csv("../data_sets/quality_of_life_indices_by_country.csv")

In [4]:
data.head()

,Rank,Country,Quality of Life Index,Purchasing Power Index,Safety Index,Health Care Index,Cost of Living Index,Property Price to Income Ratio,Traffic Commute Time Index,Pollution Index,Climate Index,Year
0,1,Switzerland,222.9,146.5,73.2,66.3,126.0,7.3,25.6,24.1,-,2015
1,2,Germany,195.9,111.8,71.5,75.6,76.3,6.6,31.1,28.3,-,2015
2,3,Sweden,193.9,110.9,57.9,76.3,82.9,9.2,26.9,15.1,-,2015
3,4,United States,192.5,126.1,50.0,67.8,76.5,2.6,36.0,31.4,-,2015
4,5,Finland,190.2,101.2,70.5,69.4,89.7,7.0,33.2,14.9,-,2015


In [5]:
data.shape

(1495, 12)

The dataset contains 12 fields.

There is one categorical field (Country), one ranking field (Rank), and ten variables describing different aspects of quality of life.

These variables include the overall Quality of Life Index, Purchasing Power Index, Safety Index, Health Care Index, Cost of Living Index, Property Price to Income Ratio, Traffic Commute Time Index, Pollution Index, Climate Index, and Year.

In [6]:
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 1495 entries, 0 to 1494
Data columns (total 12 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Rank                            1495 non-null   int64  
 1   Country                         1495 non-null   str    
 2   Quality of Life Index           1495 non-null   float64
 3   Purchasing Power Index          1495 non-null   float64
 4   Safety Index                    1495 non-null   float64
 5   Health Care Index               1495 non-null   float64
 6   Cost of Living Index            1495 non-null   float64
 7   Property Price to Income Ratio  1495 non-null   float64
 8   Traffic Commute Time Index      1495 non-null   float64
 9   Pollution Index                 1495 non-null   float64
 10  Climate Index                   1495 non-null   str    
 11  Year                            1495 non-null   str    
dtypes: float64(8), int64(1), str(3)
memory usage:

In [1]:
data.isna().sum()

NameError: name 'data' is not defined

In [8]:
data.duplicated().sum()

np.int64(0)

In [9]:
print("Non-numeric Climate Index values:", (data["Climate Index"] == "-").sum())

Non-numeric Climate Index values: 143


In [14]:
data["Year"].apply(lambda x: x.split("/")[0]).unique()

<StringArray>
['2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023',
 '2024']
Length: 10, dtype: str

In [15]:
data["Year"].unique()

<StringArray>
[  '2015', '2015/2',   '2016', '2016/2',   '2017', '2017/2',   '2018',
 '2018/2',   '2019', '2019/2',   '2020', '2020/2',   '2021', '2021/2',
   '2022', '2022/2',   '2023', '2023/2',   '2024', '2024/2']
Length: 20, dtype: str

The dataset is generally of good quality. No missing values or duplicate rows were detected.

However, some inconsistencies were found. The Year column contains values such as "2015/2", while the Climate Index column contains 143 non-numeric entries represented by "-". Because of these inconsistencies, both columns are stored as strings instead of numeric data types. These issues will be corrected during the data cleaning stage.

### Descriptive statistics

In [12]:
statistics = pd.DataFrame({
    "Mean":[data["Quality of Life Index"].mean(),
            data["Purchasing Power Index"].mean(),
            data["Safety Index"].mean(),
            data["Health Care Index"].mean(),
            data["Cost of Living Index"].mean()],

    "Median":[
        data["Quality of Life Index"].median(),
        data["Purchasing Power Index"].median(),
        data["Safety Index"].median(),
        data["Health Care Index"].median(),
        data["Cost of Living Index"].median()],

    "Standard Deviation":
        [
        data["Quality of Life Index"].std(),
        data["Purchasing Power Index"].std(),
        data["Safety Index"].std(),
        data["Health Care Index"].std(),
        data["Cost of Living Index"].std()
        ],
},
    index=["Quality of Life Index",
           "Purchasing Power Index",
           "Safety Index",
           "Health Care Index",
           "Cost of Living Index"]
)

statistics.round(2)

,Mean,Median,Standard Deviation
Quality of Life Index,134.16,136.2,40.83
Purchasing Power Index,66.80,60.7,33.28
Safety Index,60.10,59.9,13.78
Health Care Index,65.21,66.4,10.15
Cost of Living Index,53.44,49.2,20.57


Among the selected indicators, the Quality of Life Index has the highest variability, while the Health Care Index shows the lowest variability.

The mean and median values are relatively close for all indicators, suggesting that the data does not contain many extreme values.
